pkg/staking/stakingcontract/contract.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2021 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stakingcontract
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/sctx"
+	"github.com/ethersphere/bee/pkg/transaction"
+	"github.com/ethersphere/go-sw3-abi/sw3abi"
+)
+
+var (
+	MinimumStakeAmount = big.NewInt(1)
+
+	erc20ABI = parseABI(sw3abi.ERC20ABIv0_3_1)
+	//TODO: get ABI for staking contract
+	stakingABI        = parseABI(sw3abi.ERC20ABIv0_3_1)
+	stakeUpdatedTopic = stakingABI.Events["StakeUpdated"].ID
+
+	ErrInvalidStakeAmount = errors.New("invalid stake amount")
+	ErrInsufficientFunds  = errors.New("insufficient token balance")
+	ErrGetStakeFailed     = errors.New("get stake failed")
+
+	depositStakeDescription = "Deposit Stake"
+)
+
+type Interface interface {
```

Comment:
consider having other name then `Interface` since it is vary generic name. 

note: if this file is moved to `staking` package, then it might be called `Contract`